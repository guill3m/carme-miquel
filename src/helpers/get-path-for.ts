export function getPathFor(
	contentType:
		| 'article'
		| 'article-category'
		| 'book'
		| 'book-audience'
		| 'book-documents'
		| 'interview'
		| 'video'
		| 'page',
	entryId: string,
	parentEntryId?: string,
): string {
	switch (contentType) {
		case 'article':
			return `/article/${entryId}`
		case 'article-category':
			return `/articles${parentEntryId ? `/${parentEntryId}/` : '/'}${entryId}`
		case 'book':
			return `/llibre/${entryId}`
		case 'book-audience':
			return `/llibres/${entryId}`
		case 'book-documents':
			return `/llibre/${entryId}/documents`
		case 'interview':
			return `/entrevista/${entryId}`
		case 'video':
			return `/video/${entryId}`
		case 'page':
		default:
			return `/${entryId}`
	}
}
