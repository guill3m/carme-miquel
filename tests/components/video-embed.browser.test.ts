import { describe, expect, test } from 'vitest'
import { renderAstroComponent } from '../helpers'

import Link from '../../src/components/video-embed.astro'

describe('VideoEmbed component', () => {
	test('vimeo', async () => {
		const result = await renderAstroComponent(Link, {
			props: {
				aspectRatio: '16:9',
				id: 'test-id',
				platform: 'vimeo',
				title: 'Test title',
			},
		})
		const videoEmbedElement = result.firstChild
		const iframeElement = videoEmbedElement.getElementsByTagName('iframe')[0]

		expect(result).toMatchSnapshot()

		expect(iframeElement).toBeInstanceOf(HTMLIFrameElement)
		expect(iframeElement).toHaveAttribute(
			'src',
			'https://player.vimeo.com/video/test-id',
		)
		expect(iframeElement).toHaveAccessibleName('Video: Test title')
	})

	test('youtube', async () => {
		const result = await renderAstroComponent(Link, {
			props: {
				aspectRatio: '4:3',
				id: 'test-id',
				platform: 'youtube',
				title: 'Test title',
			},
		})
		const videoEmbedElement = result.firstChild
		const iframeElement = videoEmbedElement.getElementsByTagName('iframe')[0]

		expect(result).toMatchSnapshot()

		expect(iframeElement).toBeInstanceOf(HTMLIFrameElement)
		expect(iframeElement).toHaveAttribute(
			'src',
			'https://www.youtube-nocookie.com/embed/test-id',
		)
		expect(iframeElement).toHaveAccessibleName('Video: Test title')
	})

	test('youtube list', async () => {
		const result = await renderAstroComponent(Link, {
			props: {
				aspectRatio: '16:9',
				id: 'test-id',
				platform: 'youtube-list',
				title: 'Test title',
			},
		})
		const videoEmbedElement = result.firstChild
		const iframeElement = videoEmbedElement.getElementsByTagName('iframe')[0]

		expect(result).toMatchSnapshot()

		expect(iframeElement).toBeInstanceOf(HTMLIFrameElement)
		expect(iframeElement).toHaveAttribute(
			'src',
			'https://www.youtube-nocookie.com/embed/videoseries?list=test-id',
		)
		expect(iframeElement).toHaveAccessibleName('Video: Test title')
	})
})
