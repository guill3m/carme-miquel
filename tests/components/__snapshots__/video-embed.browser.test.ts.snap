// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VideoEmbed component > vimeo 1`] = `
<DocumentFragment>
  <div
    class="video ar-16-9"
    data-astro-cid-zht7hyb4=""
    data-astro-source-file="/Users/guill3m/Projects/carme-miquel/src/components/video-embed.astro"
    data-astro-source-loc="40:68"
  >
     
    <iframe
      allow="encrypted-media; fullscreen; gyroscope; picture-in-picture"
      data-astro-cid-zht7hyb4=""
      data-astro-source-file="/Users/guill3m/Projects/carme-miquel/src/components/video-embed.astro"
      data-astro-source-loc="45:3"
      src="https://player.vimeo.com/video/test-id"
      title="Video: Test title"
    >
      
	
    </iframe>
     
  </div>
   
</DocumentFragment>
`;

exports[`VideoEmbed component > youtube 1`] = `
<DocumentFragment>
  <div
    class="video ar-4-3"
    data-astro-cid-zht7hyb4=""
    data-astro-source-file="/Users/guill3m/Projects/carme-miquel/src/components/video-embed.astro"
    data-astro-source-loc="40:68"
  >
     
    <iframe
      allow="encrypted-media; fullscreen; gyroscope; picture-in-picture"
      data-astro-cid-zht7hyb4=""
      data-astro-source-file="/Users/guill3m/Projects/carme-miquel/src/components/video-embed.astro"
      data-astro-source-loc="45:3"
      src="https://www.youtube-nocookie.com/embed/test-id"
      title="Video: Test title"
    >
      
	
    </iframe>
     
  </div>
   
</DocumentFragment>
`;

exports[`VideoEmbed component > youtube list 1`] = `
<DocumentFragment>
  <div
    class="video ar-16-9"
    data-astro-cid-zht7hyb4=""
    data-astro-source-file="/Users/guill3m/Projects/carme-miquel/src/components/video-embed.astro"
    data-astro-source-loc="40:68"
  >
     
    <iframe
      allow="encrypted-media; fullscreen; gyroscope; picture-in-picture"
      data-astro-cid-zht7hyb4=""
      data-astro-source-file="/Users/guill3m/Projects/carme-miquel/src/components/video-embed.astro"
      data-astro-source-loc="45:3"
      src="https://www.youtube-nocookie.com/embed/videoseries?list=test-id"
      title="Video: Test title"
    >
      
	
    </iframe>
     
  </div>
   
</DocumentFragment>
`;
